scipy/optimize/tests/test__remove_redundancy.py

```diff
@@ -16,6 +16,9 @@
 
 from .test_linprog import magic_square
 from scipy.optimize._remove_redundancy import _remove_redundancy
+from scipy.optimize._remove_redundancy import _remove_redundancy_dense
+from scipy.optimize._remove_redundancy import _remove_redundancy_sparse
```

Comment:
A quick question, there are now three routines ``_remove_redundancy``, ``_remove_redundancy_dense`` and ``_remove_redundancy_sparse``, obviously I would use the dense routine for dense arrays and the sparse routine for sparse arrays. But under what circumstances would I use the general ``_remove_redundancy`` over the dense or sparse versions?

Reply:
They are just poorly named. `_remove_redundancy` is another dense routine that can be useful if the dimension of the nullspace is small. If you can merge this, gh-11249 would be a perfect place to fix the names.

Reply:
Ok I'll merge it now.